### ISSUE TITLE
Build Chain / agent queue

### ISSUE THREAD

**Issue by Sebizbir** [megoels/VSTS-Extensions#1]
Hi, 
Is there some factor like Demand (Phase 1) , Variable (Variable tab) on particular build I can set to do below:

When one agent have queue lenght (there are more than one build assigned) of 2, run next agent. Meaning, if one agent is doing a build and next build is awaiting, run second agent(vm).

This setup is for VM scaling where private agents runns.
Right now situation is either one agent (on VM1) have many build queued or I need to launch few VMs manually.

But I need VM scaling based on VM1 queue, this way next VMs could launch automatically (auto shutdown/deallocate script does closing job).

Maybe you know easier way (REST API inst giving clear results).

Please advice.

Regards Seb

**Comment by grawcho** [comment]
megoels - you can add the Disable / Enable task we worked on and this could support the requirement.
however, the queue length is a REST call that needs to happen with a listener and i don't think that the place to implement this is a build task. you might need to setup a webApp or use Azure functions or logical apps to perform that queue availability check during the build as a listener while the build is running.
bottom line ... this is not the place to implement it IMHO.
G.

**Comment by megoels** [comment]
Hi and thank you,

This sounds very useful, i'm agree with grawcho this suppose to be kind of listener to the Agent queue which allocating and deallocating VM's according to usage.
I think it kind of extension which could be added to the Pool queue and allow you to configure and  monitor the queue to allocate/deallocate VM's.
Regarding the Disable/Enable Agents it included on this extension.

Regards.

**Comment by Sebizbir** [comment]
Ok then,
Should I wait you guys add agent queue feature to this extension?

Cheers

**Comment by megoels** [comment]
I've suggested new extension which can be developed and as grawcho mentioned it need to be listener,
Nothing to add to this extension which is dedicated to build chain so your suggestion isn't applicable to this extension !
